pkgs/tools/virtualization/mininet/default.nix

```diff
@@ -0,0 +1,46 @@
+{ stdenv, lib, fetchFromGitHub
+, which
+, python
+, help2man
+}:
+
+let
+  pyEnv = python.withPackages(ps: [ ps.setuptools ]);
+in
+stdenv.mkDerivation rec {
+  name = "mininet-${version}";
+  version = "2.3.0d4";
+
+  outputs = [ "out" "py" ];
+
+  src = fetchFromGitHub {
+    owner = "mininet";
+    repo = "mininet";
+    rev = version;
+    sha256 = "02hsqa7r5ykj8m1ycl32xwn1agjrw78wkq87xif0dl2vkzln41i4";
+  };
+
+  buildFlags = [ "mnexec" ];
+  makeFlags = [ "PREFIX=$(out)" ];
+
+  installTargets = [ "install-mnexec" "install-manpages" ];
+
+  preInstall = ''
+    mkdir -p $out $py
+    # without --root, install fails
+    ${pyEnv.interpreter} setup.py install --root="/" --prefix=$py
+  '';
+
+  doCheck = false;
+
+  buildInputs = [ python.pkgs.wrapPython which help2man pyEnv ];
```

Comment:
If you add `setuptools` as pythonPath, `wrapPython` should pick it up.
See also https://github.com/NixOS/nixpkgs/blob/master/pkgs/build-support/rust/fetchcargo.nix#L8

Reply:
But is `setuptools` really needed at runtime?
If it is only used at build time, then adding `setuptools` to `buildInputs` is all it takes.

Reply:
the makefile overrides PYTHONPATH which generates an error. In principle it should be needed only during install. I will try to send a patch.

Reply:
nevermind, the problem was something else. Is it ok now ?